exec/vector/src/main/java/org/apache/drill/exec/record/MaterializedField.java

```diff
@@ -115,56 +175,17 @@ public void removeChild(MaterializedField field) {
    */
 
   public void replaceType(MajorType newType) {
-    assert type.getMinorType() == newType.getMinorType();
-    assert type.getMode() == newType.getMode();
+    Preconditions.checkArgument(type.getMinorType() == newType.getMinorType());
+    Preconditions.checkArgument(type.getMode() == newType.getMode());
```

Comment:
Any reason to pay this cost for all queries? Isn't a test-time assert sufficient?

Also, what the heck does it mean to replace a type if we assert that the new type is the same as the old one? We are actually setting the precision and scale fields. Maybe the child types.

Reply:
I would be concerned with the cost only in case `replaceType()` is called for every row (tuple) (note that in java there is no test-time assert, asserts are optimized out only when bytecode is compiled by the hotspot) and I have no idea what it means to replace a type.